lib/matplotlib/dates.py

```diff
@@ -201,6 +220,65 @@ def _get_rc_timezone():
     MO, TU, WE, TH, FR, SA, SU)
 WEEKDAYS = (MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY, SUNDAY)
 
+# default epoch: passed to np.datetime64...
+_epoch = None
+_epoch_used = False
```

Comment:
Do we need two variables? Can't we fetch `rcParams['date.epoch']` on first usage and check `_epoch is not None` instead of `_epoch_uesd`?

Reply:
This is the latch that causes an error if `date2num` has been called, (i.e. `get_epoch` called).  Added a comment....

Reply:
My question is: Can't you just just
~~~
epoch = None

def set_epoch(epoch):
    global _epoch
    if _epoch is not None:
        raise RuntimeError('set_epoch must be called before dates plotted.')
    _epoch = epoch

def get_epoch(epoch):
    global _epoch
    if _epoch is None:
        _epoch = matplotlib.rcParams['date.epoch']
    return _epoch
~~~
without the need for  `_epoch_used`.

Reply:
Yes agreed, that is better... Thanks!